### PR TITLE
Change to use async methods instead of deprecated sync methods.

### PR DESCRIPTION
**Pull request by kaituo**
Note: since there are inter-file references, I split a big PR into multiple smaller PRs based on files to save time (1.1 cut off is due at the end of this week). The build will fail due to missing dependencies. I will use that PR just for review. will not merge it. Will have a big PR in the end and merge once after all review PRs get approved. For the complete code, check https://github.com/kaituo/anomaly-detection-1/tree/compactRCF2

### Description

Testing done:
1. Manually verified the change does not impact functionality.
 
### Issues Resolved
https://github.com/opensearch-project/anomaly-detection/issues/97
 
### Check List
- [ x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/ad/transport/DeleteModelTransportAction.java [hunk 1]
```diff
@@ -113,7 +114,15 @@ protected DeleteModelNodeResponse nodeOperation(DeleteModelNodeRequest request)
...
         LOG.info("Delete model for {}", adID);
         // delete in-memory models and model checkpoint
-        modelManager.clear(adID);
+        modelManager
+            .clear(
+                adID,
+                ActionListener
+                    .wrap(
+                        r -> LOG.info("Deleted model for [{}] with response [{}] ", adID, r),
+                        e -> LOG.error("fail to delete model for " + adID, e)
```
Comment (ohltyler) [inline src/main/java/org/opensearch/ad/transport/DeleteModelTransportAction.java:123]: super minor: capitalize to `Fail` for consistency

**Review by ohltyler — DISMISSED** [review]

File: src/main/java/org/opensearch/ad/transport/DeleteModelTransportAction.java (hunk 1 above)
Comment (kaituo) [reply src/main/java/org/opensearch/ad/transport/DeleteModelTransportAction.java:123]: yeah, changed

File: src/main/java/org/opensearch/ad/transport/DeleteModelTransportAction.java [hunk 2]
```diff
@@ -113,7 +114,15 @@ protected DeleteModelNodeResponse nodeOperation(DeleteModelNodeRequest request)
         String adID = request.getAdID();
         LOG.info("Delete model for {}", adID);
         // delete in-memory models and model checkpoint
-        modelManager.clear(adID);
+        modelManager
+            .clear(
+                adID,
+                ActionListener
+                    .wrap(
+                        r -> LOG.info("Deleted model for [{}] with response [{}] ", adID, r),
```
Comment (ylwu-amzn) [inline src/main/java/org/opensearch/ad/transport/DeleteModelTransportAction.java:122]: When will we call this delete model action? Will it cause too many log lines for HC detector?

Comment (kaituo) [reply src/main/java/org/opensearch/ad/transport/DeleteModelTransportAction.java:122]: We call it when people stop or delete a detector.  So it should not cause too many log lines, right?

**Review by ohltyler — APPROVED** [review]

File: src/main/java/org/opensearch/ad/transport/DeleteModelTransportAction.java (hunk 2 above)
Comment (ylwu-amzn) [reply src/main/java/org/opensearch/ad/transport/DeleteModelTransportAction.java:122]: Yes, that's ok.

**Review by ylwu-amzn — APPROVED** [review]

**Pull request closed without merging**